lib/parameters.py

```diff
@@ -181,6 +181,30 @@ def gen_avc_vbr_la_parameters(spec, profiles):
   params = gen_avc_vbr_la_variants(spec, profiles)
   return keys, params
 
+def gen_avc_forced_idr_variants(spec, profiles, tested):
+  for case, params in spec.items():
+    for variant in copy.deepcopy(params.get("forced_idr", [])):
+      uprofile = variant.get("profile", None)
+      cprofiles = [uprofile] if uprofile else profiles
+      rcmode = variant.get("rcmode", None)
+      variant.update(
+        bitrate = None if "cqp" == rcmode else variant["bitrate"],
+        maxrate = None if "cqp" == rcmode else (variant["bitrate"] if "cbr" == rcmode else variant["bitrate"] * 2),
+        qp = None if "cqp" != rcmode else variant["qp"])
```

Comment:
I think it is better to assert this logic instead of silently ignoring/overriding the user test config.  For example, if user test config sets `rcmode = "cqp"` with a `bitrate`, then let it throw an exception because it is an invalid test config definition.